### PR TITLE
[chg] remove drift detection for EventSubscription's tag

### PR DESCRIPTION
**Pull request by leo19900723**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ZiruiHua94 — APPROVED** [review]

File: aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java [hunk 1]
```diff
@@ -183,15 +184,8 @@ static ResourceModel translateFromReadTagsResponse(final DescribeTagsResponse aw
...
-                .stream()
-                .filter(tag -> currentResourceState.getTags() == null || !currentResourceState.getTags().contains(tag))
-                .collect(Collectors.toList());
-
-        List<Tag> toBeDeletedTags = currentResourceState.getTags() == null ? Collections.emptyList() : currentResourceState.getTags()
-                .stream()
-                .filter(tag -> desiredResourceState.getTags() == null || !desiredResourceState.getTags().contains(tag))
-                .collect(Collectors.toList());
+        List<Tag> toBeCreatedTags = subtract(desiredResourceState.getTags(), currentResourceState.getTags());
+        List<Tag> toBeDeletedTags = subtract(currentResourceState.getTags(), desiredResourceState.getTags());
```
Comment (shwetayakkali) [inline aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java:188]: this implies set difference ?

File: aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java [hunk 2]
```diff
@@ -212,21 +206,44 @@ private static software.amazon.awssdk.services.redshift.model.Tag translateToSdk
         return GSON.fromJson(GSON.toJson(tag), software.amazon.awssdk.services.redshift.model.Tag.class);
     }
 
-    public static List<software.amazon.awssdk.services.redshift.model.Tag> translateToSdkTags(List<Tag> tags) {
-        return tags == null ? null : tags
-                .stream()
-                .map(Translator::translateToSdkTag)
-                .collect(Collectors.toList());
+    public static List<software.amazon.awssdk.services.redshift.model.Tag> translateToSdkTags(List<Tag> modelTags) {
```
Comment (shwetayakkali) [inline aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java:209]: have you combine refactoring of Null checks in this PR?

File: aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java (hunk 1 above)
Comment (leo19900723) [reply aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java:188]: I implement `subtract` method to replace this part in order to reduce code duplicity. The fundamental logic is the same. I've already implemented this part in ClusterParameterGroup: [here](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-redshift/blob/master/aws-redshift-clusterparametergroup/src/main/java/software/amazon/redshift/clusterparametergroup/Translator.java#L285-L302) and it works for months.

File: aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java (hunk 2 above)
Comment (leo19900723) [reply aws-redshift-eventsubscription/src/main/java/software/amazon/redshift/eventsubscription/Translator.java:209]: yes

**Review by shwetayakkali — APPROVED** [review]

**Pull request merged**